soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -176,8 +188,11 @@ __asm__(".align 4\n\t"
 	"  wsr   a0, WINDOWBASE\n\t"
 	"  rsync\n\t"
 	"  movi  sp, d0i3_stack\n\t"
-	"  movi a2, 0x1000\n\t"
-	"  add sp, sp, a2\n\t"
+	"  rsr   a2, PRID\n\t"
+	"  movi  a3, 0x400\n\t"
+	"  mull  a2, a2, a3\n\t"
+	"  add   a2, a2, a3\n\t"
+	"  add   sp, sp, a2\n\t"
```

Comment:
What's going on here?  You're computing ((PRID + 1) * 1024) and popping that many bytes off the stack?  It was weird enough before where there seems to have been a 4k buffer being adjusted, but I don't understand at all why the stack adjustment needs to be CPU-specific.  This is the CPU-local interrupt stack, isn't it?

Reply:
OK, now I get it.  This is just computing the end of the relevant subarray, and doing the computation at runtime because the __asm__ block is global and so can't feed C expressions via constraints.  That's annoying, I've been there.  But can you clean this up so you aren't hardcoding 0x400 == CONFIG_IDLE_STACK_SIZE, or at the very least put a comment here explaining what's going on?

Still, I don't get the point here.  Why are you wasting 4k (!!) of precious HPSRAM for per-core power stacks, when there's a perfectly good and completely unused (because interrupts are masked, right?  right?!) per-core interrupt stack already.

Please find a way to unify this with the interrupt stack.